libflux/flux-core/src/semantic/types.rs

```diff
@@ -261,6 +261,9 @@ impl cmp::PartialOrd for Kind {
     }
 }
 
+/// Pointer type used in `MonoType`
+pub type Ptr<T> = Box<T>;
```

Comment:
This is orthogonal to the main change in this PR but I initially wanted to check how difficult it would be to use `Arc` instead to allow types to be shared instead of duplicated (via clone). However due to `Substitutable` and `Fresh` taking `self` by value that wasn't possible straight away but will instead come in a separate PR (which further reduces the runtime).

Still left it in as it would conflict with the other changes.